### PR TITLE
Nonce handling by hand

### PR DESCRIPTION
**Pull request by kostekIV**

### CODE REVIEW

File: .github/workflows/on-master-branch-commit.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
   push:
     branches:
       - master
+      - nonce
```
Comment (fbielejec) [inline .github/workflows/on-master-branch-commit.yml:8]: don;t forget to remove!

File: relayer/relayer/Cargo.toml [hunk 2]
```diff
@@ -23,3 +23,4 @@ signer_client = { path = "../signer_client" }
 subxt = { workspace = true }
 thiserror = "1.0.48"
 tokio = { workspace = true, features = ["rt", "rt-multi-thread", "macros", "test-util", "sync"] }
+tracing = { workspace = true, features = ["log", "log-always"] }
```
Comment (fbielejec) [inline relayer/relayer/Cargo.toml:26]: I'm adding it in another PR,  but cool, lets keep it here

File: .github/workflows/on-master-branch-commit.yml (hunk 1 above)
Comment (kostekIV) [reply .github/workflows/on-master-branch-commit.yml]: done

File: relayer/relayer/src/connections/azero.rs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+use std::sync::{Arc, Mutex as StdMutex};
```
Comment (fbielejec) [inline relayer/relayer/src/connections/azero.rs:1]: I would called SyncMutex (since Tokio is Async/await)

Comment (kostekIV) [reply relayer/relayer/src/connections/azero.rs]: done

File: relayer/relayer/src/connections/azero.rs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+use std::sync::{Arc, Mutex as SyncMutex};
```
Comment (krzysztofziobro) [inline relayer/relayer/src/connections/azero.rs:1]: Is there some good reason here to use Mutex from std instead of tokio Mutex?

Comment (kostekIV) [reply relayer/relayer/src/connections/azero.rs:1]: 2nd case I think we should only increase the nonce if we were successful in submitin it to the chain. I can use the tokio mutex, np

**Review by fbielejec — APPROVED** [review]
You could remove some of the duplication by creating a `with_nonce` decorator that augments the existing type, instead of adding the nonce to the `with_keypair`, but generally LGTM

File: relayer/relayer/src/connections/azero.rs (hunk 4 above)
Comment (fbielejec) [reply relayer/relayer/src/connections/azero.rs:1]: I think the sync version has less overhead, even in async scenarios (read some articles on that).

File: relayer/relayer/src/connections/azero.rs (hunk 3 above)
Comment (kostekIV) [reply relayer/relayer/src/connections/azero.rs]: changed to atomicu64

File: relayer/relayer/src/connections/azero.rs (hunk 4 above)
Comment (kostekIV) [reply relayer/relayer/src/connections/azero.rs]: after discusion changed it to Atomicu64

**Comment by kostekIV** [comment]
> You could remove some of the duplication by creating a `with_nonce` decorator that augments the existing type, instead of adding the nonce to the `with_keypair`, but generally LGTM

I dont like macros etc :P So I have just created new common method.

**Comment by fbielejec** [comment]
> > You could remove some of the duplication by creating a `with_nonce` decorator that augments the existing type, instead of adding the nonce to the `with_keypair`, but generally LGTM
> 
> I dont like macros etc :P So I have just created new common method.

You could do it w-out any macros.. but this works equally fine

**Pull request merged**